### DISCUSSION TITLE
reverse proxy name resolution

### DISCUSSION THREAD

**Discussion by gonsalvesc** [dani-garcia/vaultwarden#4363] [Q&A] [answered]
Hello Vaultwarden community,
I have installed a private instance of Vaultwarden that can be accessed only on my local network via Caddy reverse proxy, and it works perfectly!
My question is about the reverse_proxy directive in Caddyfile. Using the below directive, how does Caddy know to resolve "vaultwarden" to the right IP? I have looked through all the files (compose, .env, Caddyfile) and there is nothing that helps resolve this!

`reverse_proxy vaultwarden:8000`

**Comment by stefan0xC** [comment, marked as answer]
> how does Caddy know to resolve "vaultwarden" to the right IP?

This is done automagically by [Docker Compose](https://docs.docker.com/compose/networking/): "_By default Compose sets up a single [network](https://docs.docker.com/engine/reference/commandline/network_create/) for your app. Each container for a service joins the default network and is both reachable by other containers on that network, and discoverable by the service's name."_

> I have looked through all the files (compose, .env, Caddyfile) and there is nothing that helps resolve this!

Well, as mentioned by the docs there is:
```yaml
services:
  vaultwarden:
```

To dispell the magic (and assuming you are using a `docker-compose.yml` like [the example in the wiki](https://github.com/dani-garcia/vaultwarden/wiki/Using-Docker-Compose)) you can run something like the following to see how the configuration will be used to setup the networking inside the `caddy` container:

```bash
docker exec caddy cat /etc/hosts
```

Note: The `container_name` will depend on your configuration. The `caddy` corresponds to the `container_name: caddy` directive of the `caddy` service (check `docker ps` to list the running containers in case you are unsure) and the `vaultwarden` in the hosts file should correspond to the `vaultwarden` service (as explained above).

**Comment by gonsalvesc** [comment]
Thank you so much for the detailed reply! Totally did not know you could use the service name like that! Learn something new everyday!!
